packages/core/src/app.ts

```diff
@@ -1524,11 +1536,36 @@ export class App {
         }
     ) {
         const { vault } = this.getItem(item.id)!;
-        vault.items.update(new VaultItem({ ...item, ...upd, updatedBy: this.account!.id }));
+        const newItem = new VaultItem({ ...item, ...upd, updatedBy: this.account!.id });
+        this._updateHistory(item, newItem);
+        vault.items.update(newItem);
         await this.saveVault(vault);
         await this.syncVault(vault);
     }
 
+    protected _updateHistory(oldItem: VaultItem, newItem: VaultItem) {
+        const newHistoryEntry = new ItemHistory();
+
+        newHistoryEntry.updatedBy = newItem.updatedBy;
+
+        if (oldItem.name !== newItem.name) {
+            newHistoryEntry.name = oldItem.name;
+        }
+
+        if (JSON.stringify(oldItem.tags) !== JSON.stringify(newItem.tags)) {
+            newHistoryEntry.tags = oldItem.tags;
+        }
+
+        if (JSON.stringify(oldItem.fields) !== JSON.stringify(newItem.fields)) {
+            newHistoryEntry.fields = oldItem.fields;
+        }
+
+        if (newHistoryEntry.name || newHistoryEntry.tags || newHistoryEntry.fields) {
+            // Prepend (first is the most recent change) and cap
+            newItem.history = [newHistoryEntry, ...newItem.history].slice(0, ITEM_HISTORY_ENTRIES_LIMIT);
+        }
+    }
```

Comment:
I think every history entry should contain every all of these properties, even if they haven't changed. This may take up a bit more space but makes restoring a state much simpler. In fact, I would argue that it's necessary. Consider the following scenario:

An item was changed > 10 times. Now the user wants to restore the last item the history. No fields had changed in that particular instance, but they did change later. Because the fields are not stored in the last history entry, end no older entries are available, the fields can not be restored to that point in time. So we either need a before and after for each of these properties or we'll simply store all properties in every history entry. I would argue that the latter option is much simpler and more practical, even if it potentially takes up a bit more space.